<a href="https://colab.research.google.com/github/evaneliz/534groupprojfiles/blob/main/EScopygroup_assignment_5_evanpopparams.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
!pip install numpy pandas seaborn

In [2]:
import numpy as np
import pandas as pd
import seaborn as sns
from scipy import stats

In [3]:
import geopandas as gpd
import pandas as pd
from shapely.geometry import Point

In [4]:
sns.set_style("whitegrid")
sns.set_context("paper")

np.random.seed(0)

In [5]:
url = 'https://raw.githubusercontent.com/evaneliz/534groupprojfiles/refs/heads/main/RECLEANEDsocial_explorer_census_data%20-%20RECLEANEDsocial_explorer_census_data.csv'
dftransit = pd.read_csv(url)

In [6]:
dftransit.head()

Statistics    2080.01    2083.01    2083.02    2084.01    2085.02  \
0          total_pop      2,452      1,904      4,881      4,092      4,208   
1        pop_density  11,025.90  53,050.80  28,733.20  46,144.30  41,750.50   
2               area       0.22       0.04       0.17       0.09        0.1   
3      total_workers      1,828        961      2,248      2,154      2,011   
4  pct_car_truck_van     59.10%     45.10%     62.50%     52.90%     56.90%   

      2087.1     2087.2    2088.01    2088.02  ...     2098.2     2100.1  \
0      4,757      3,917      3,397      2,792  ...      2,559      3,349   
1  72,312.30  61,583.30  37,559.80  34,191.10  ...  37,004.60  21,208.90   
2       0.07       0.06       0.09       0.08  ...       0.07       0.16   
3      2,904      2,070      1,507      1,473  ...      1,230      1,516   
4     62.80%     43.60%     54.90%     59.60%  ...     62.30%     45.40%   

     2111.23    2111.24    2122.02    2122.03    2122.04    2134.01  \
0      1,748      4,321      3,417      2,733      3,431      4,818   
1  26,316.60  28,329.40  42,863.10  66,581.70  41,683.30  63,633.70   
2       0.07       0.15       0.08       0.04       0.08       0.08   
3      1,135      2,291      1,947      1,366      1,801      2,586   
4     46.90%     56.70%     52.60%     51.80%     64.10%     70.30%   

     2134.02      TOTAL  
0      4,697    115,435  
1  31,045.80  40,539.50  
2       0.15       2.85  
3      2,290     57,498  
4     68.90%     52.20%  

[5 rows x 36 columns]

In [7]:
dftransit["Statistics"] = dftransit["Statistics"].str.replace(".", "", regex=False)

dftransit["Statistics"].astype(str)

0                    total_pop
1                  pop_density
2                         area
3                total_workers
4            pct_car_truck_van
5              pct_drove_alone
6                pct_carpooled
7               pct_pubtransit
8               pct_motorcycle
9                     pct_bike
10                 pcty_walked
11                   pct_other
12                     pct_wfh
13                  pct_nonwfh
14           pct_less_than_ten
15         pct_ten_to_nineteen
16    pct_twenty_to_twentynine
17    pct_thirty_to_thirtynine
18      pct_forty_to_fiftynine
19     pct_sixty_to_eightynine
20          pct_ninety_or_more
Name: Statistics, dtype: object

In [8]:
for column in dftransit.columns:
  if "pct" in column.lower():
    dftransit[column] = dftransit[column].str.replace('%', '', regex=False).astype(float)

In [9]:
dftransit = dftransit.T

In [10]:
dftransit.rename(columns=dftransit.iloc[0], inplace=True)

In [11]:
dftransit.drop(dftransit.index[0], inplace=True)

In [12]:
dftransit = dftransit.reset_index()

In [13]:
dftransit.head()

index total_pop pop_density  area total_workers pct_car_truck_van  \
0  2080.01     2,452   11,025.90  0.22         1,828            59.10%   
1  2083.01     1,904   53,050.80  0.04           961            45.10%   
2  2083.02     4,881   28,733.20  0.17         2,248            62.50%   
3  2084.01     4,092   46,144.30  0.09         2,154            52.90%   
4  2085.02     4,208   41,750.50   0.1         2,011            56.90%   

  pct_drove_alone pct_carpooled pct_pubtransit pct_motorcycle  ... pct_other  \
0          41.10%        18.00%         14.60%          0.00%  ...     0.00%   
1          34.40%        10.60%         39.70%          0.00%  ...     1.50%   
2          52.60%        10.00%         22.80%          0.00%  ...     1.70%   
3          39.10%        13.80%         42.50%          0.00%  ...     1.10%   
4          48.40%         8.50%         27.80%          0.10%  ...     1.10%   

  pct_wfh pct_nonwfh pct_less_than_ten pct_ten_to_nineteen  \
0  11.30%     88.70%             2.60%              30.40%   
1   6.20%     93.80%             5.00%              12.20%   
2   8.50%     91.50%             5.30%              13.60%   
3   2.50%     97.50%             2.40%              17.80%   
4   3.70%     96.30%             4.50%              22.60%   

  pct_twenty_to_twentynine pct_thirty_to_thirtynine pct_forty_to_fiftynine  \
0                   13.20%                   22.80%                 14.40%   
1                   13.60%                   26.30%                 17.50%   
2                   12.40%                   27.30%                 13.50%   
3                   14.50%                   35.90%                  7.50%   
4                   13.30%                   23.90%                 17.90%   

  pct_sixty_to_eightynine pct_ninety_or_more  
0                   5.30%              0.00%  
1                  14.80%              4.40%  
2                  11.80%              7.70%  
3                  13.00%              6.40%  
4                   6.20%              7.80%  

[5 rows x 22 columns]

In [14]:
url = "https://raw.githubusercontent.com/evaneliz/534groupprojfiles/refs/heads/main/indexCensusZonalStatsMP.08072022.csv"
dfheat = pd.read_csv(url)

In [15]:
dfheat.head()

STATEFP  COUNTYFP  TRACTCE              AFFGEOID       GEOID    index  \
0        6        37   212203  1400000US06037212203  6037212203  2122.03   
1        6        37   209104  1400000US06037209104  6037209104  2091.04   
2        6        37   212204  1400000US06037212204  6037212204  2122.04   
3        6        37   212202  1400000US06037212202  6037212202  2122.02   
4        6        37   209820  1400000US06037209820  6037209820  2098.20   

               NAMELSAD STUSPS          NAMELSADCO  STATE_NAME LSAD   ALAND  \
0  Census Tract 2122.03     CA  Los Angeles County  California   CT  106312   
1  Census Tract 2091.04     CA  Los Angeles County  California   CT   69958   
2  Census Tract 2122.04     CA  Los Angeles County  California   CT  213185   
3  Census Tract 2122.02     CA  Los Angeles County  California   CT  206471   
4  Census Tract 2098.20     CA  Los Angeles County  California   CT  179107   

   AWATER       _mean     _median    _stdev        _min        _max    _range  
0       0  319.431539  319.410004  0.757958  317.760010  320.779999  3.019989  
1       0  320.068232  320.160004  0.577502  318.899994  321.059998  2.160004  
2       0  319.787408  319.869995  0.765644  318.000000  320.899994  2.899994  
3       0  320.469810  320.440002  0.433844  319.559998  321.540009  1.980011  
4       0  320.675219  320.630005  0.494390  319.559998  321.980011  2.420013

In [16]:
dftransit = dftransit[dftransit['index'] != "TOTAL"]

In [17]:
dftransit['index'] = pd.to_numeric(dftransit['index'])

<ipython-input-17-671ab3652670>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dftransit['index'] = pd.to_numeric(dftransit['index'])


In [18]:
merged_df = pd.merge(dftransit, dfheat, on='index')
print(merged_df.head())

     index total_pop pop_density  area total_workers pct_car_truck_van  \
0  2080.01     2,452   11,025.90  0.22         1,828            59.10%   
1  2083.01     1,904   53,050.80  0.04           961            45.10%   
2  2083.02     4,881   28,733.20  0.17         2,248            62.50%   
3  2084.01     4,092   46,144.30  0.09         2,154            52.90%   
4  2085.02     4,208   41,750.50   0.1         2,011            56.90%   

  pct_drove_alone pct_carpooled pct_pubtransit pct_motorcycle  ...  \
0          41.10%        18.00%         14.60%          0.00%  ...   
1          34.40%        10.60%         39.70%          0.00%  ...   
2          52.60%        10.00%         22.80%          0.00%  ...   
3          39.10%        13.80%         42.50%          0.00%  ...   
4          48.40%         8.50%         27.80%          0.10%  ...   

   STATE_NAME LSAD   ALAND AWATER       _mean     _median    _stdev  \
0  California   CT  575976      0  318.805957  318.739990  1.11

In [19]:
merged_df['pop_density'] = merged_df['pop_density'].astype(str)

In [20]:
merged_df['pop_density'] = merged_df['pop_density'].str.replace(',', '').str.split(',').str[0]

In [21]:
merged_df['pop_density'] = pd.to_numeric(merged_df['pop_density'])

In [22]:
#creating population parameters
# Calculate the mean of 'pop_density'
population_mean = merged_df['pop_density'].mean()

# Calculate the standard deviation of 'pop_density'
population_std = merged_df['pop_density'].std()

# Calculate the sample size
sample_size = len(merged_df['pop_density'])

# Calculate the standard error of the mean
standard_error = population_std / np.sqrt(sample_size)


print(f"Population Mean: {population_mean}")
print(f"Population Standard Deviation: {population_std}")
print(f"Sample Size: {sample_size}")
print(f"Standard Error: {standard_error}")

# Example: Calculate a 95% confidence interval for the population mean
confidence_level = 0.95
z_score = stats.norm.ppf((1 + confidence_level) / 2)
margin_of_error = z_score * standard_error
confidence_interval = (population_mean - margin_of_error, population_mean + margin_of_error)

print(f"{confidence_level*100:.0f}% Confidence Interval: {confidence_interval}")

Population Mean: 48738.99705882353
Population Standard Deviation: 22839.36426253274
Sample Size: 34
Standard Error: 3916.9186565787445
95% Confidence Interval: (41061.97756155618, 56416.01655609088)


**Interpretation & Analysis of Population Parameters**

In this inferential analysis, we examined population densite in Macarthur Park, our area of focus, to establish a base understanding of the area's demographic concentration. Estimating population parameters provides important and much needed context for our research. Population density is important in analyzing transportation behavior, especially considering higher densities often have a greater reliance on public transit. Understanding the pop_density distribution allows us to better interpret correlations with transit choices under different temperature conditions.

**Inductive Interpretation:**



*   Pop Mean: 48,739 people/square mile: This number suggests that MacArthur Park is a highly dense, urbanized neighborhood. Which is consistent with our research.
*   Std Deviation: The number 22,839 shows that there is significant variability in density between different zones. This is consistent with what we know of the neighborhood area outside of the data since the area consists of commercial hubs, residential zones, and the park itself--all of which would have varying pop densities.
*   Std Error: 3,917, The smaller number indicates that the sample mean is reliable and the expected variation is slim.

Overall, the numerical data confirms that MacArthur Park is a densely populated area with its respective zones varying in density.


In [23]:
import statsmodels.api as sm

In [24]:
gdf_tracts = gpd.read_file(
    "https://raw.githubusercontent.com/evaneliz/534groupprojfiles/refs/heads/main/censustractmparkshape.shp"
)
gdf_tracts.shape

(34, 14)

In [25]:
print(merged_df.columns)

Index(['index', 'total_pop', 'pop_density', 'area', 'total_workers',
       'pct_car_truck_van', 'pct_drove_alone', 'pct_carpooled',
       'pct_pubtransit', 'pct_motorcycle', 'pct_bike', 'pcty_walked',
       'pct_other', 'pct_wfh', 'pct_nonwfh', 'pct_less_than_ten',
       'pct_ten_to_nineteen', 'pct_twenty_to_twentynine',
       'pct_thirty_to_thirtynine', 'pct_forty_to_fiftynine',
       'pct_sixty_to_eightynine', 'pct_ninety_or_more', 'STATEFP', 'COUNTYFP',
       'TRACTCE', 'AFFGEOID', 'GEOID', 'NAMELSAD', 'STUSPS', 'NAMELSADCO',
       'STATE_NAME', 'LSAD', 'ALAND', 'AWATER', '_mean', '_median', '_stdev',
       '_min', '_max', '_range'],
      dtype='object')


In [26]:
print(gdf_tracts['NAME'].dtype)
print(merged_df['index'].dtype)

object
float64


In [27]:
merged_df['index'] = merged_df['index'].astype(str)

In [28]:
gdf_tracts_data = pd.merge(
    left=gdf_tracts, right=merged_df, left_on="NAME", right_on="index", how="left"
)
gdf_tracts_data.shape

(34, 54)

In [29]:
gdf_tracts_data.columns

Index(['STATEFP_x', 'COUNTYFP_x', 'TRACTCE_x', 'AFFGEOID_x', 'GEOID_x', 'NAME',
       'NAMELSAD_x', 'STUSPS_x', 'NAMELSADCO_x', 'STATE_NAME_x', 'LSAD_x',
       'ALAND_x', 'AWATER_x', 'geometry', 'index', 'total_pop', 'pop_density',
       'area', 'total_workers', 'pct_car_truck_van', 'pct_drove_alone',
       'pct_carpooled', 'pct_pubtransit', 'pct_motorcycle', 'pct_bike',
       'pcty_walked', 'pct_other', 'pct_wfh', 'pct_nonwfh',
       'pct_less_than_ten', 'pct_ten_to_nineteen', 'pct_twenty_to_twentynine',
       'pct_thirty_to_thirtynine', 'pct_forty_to_fiftynine',
       'pct_sixty_to_eightynine', 'pct_ninety_or_more', 'STATEFP_y',
       'COUNTYFP_y', 'TRACTCE_y', 'AFFGEOID_y', 'GEOID_y', 'NAMELSAD_y',
       'STUSPS_y', 'NAMELSADCO_y', 'STATE_NAME_y', 'LSAD_y', 'ALAND_y',
       'AWATER_y', '_mean', '_median', '_stdev', '_min', '_max', '_range'],
      dtype='object')

In [30]:
print(gdf_tracts_data['pop_density'].dtype)
print(gdf_tracts_data['_max'].dtype)

float64
float64


In [31]:
response = "pop_density"
predictor = "_max"

In [32]:
data = gdf_tracts_data[[response, predictor]].dropna()
print(data.shape)


X = data[predictor]
y = data[response]

(26, 2)


In [33]:
model = sm.OLS(y, sm.add_constant(X))
result = model.fit()
print(result.summary())

                            OLS Regression Results                            
Dep. Variable:            pop_density   R-squared:                       0.001
Model:                            OLS   Adj. R-squared:                 -0.041
Method:                 Least Squares   F-statistic:                   0.02351
Date:                Wed, 27 Nov 2024   Prob (F-statistic):              0.879
Time:                        19:49:11   Log-Likelihood:                -299.06
No. Observations:                  26   AIC:                             602.1
Df Residuals:                      24   BIC:                             604.6
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const      -1.354e+05   1.21e+06     -0.112      0.9



Evan McAvenia = uploaded and cleaned data, created population parameters
Emily Sanchez = Explained population parameter results.